### PR TITLE
thread pinning

### LINKED ISSUE #75: Thread pinning
We should be able to execute, from the run control, a thread pinning script in readoutlibs:
https://github.com/DUNE-DAQ/readoutlibs/tree/develop/scripts
This should be executed at the correct time, and automated.

### PR DESCRIPTION
**Pull request by plasorak**
This is an attempt to fix this: https://github.com/DUNE-DAQ/nanorc/issues/75
However, this is not automated yet.

### CODE REVIEW

**Comment by plasorak** [comment]
@philiprodrigues, if you could test this, that'd be good.

The behaviour depends on whether we use `nanorc` or `nano04rc`:
 - If one uses `nanorc`, thread pinning is disabled by default, and one can pass the following arguments to conf: `--enable-thread-pinning`, `--thread-pinning-file`, `--readout-app`. The thread pinning file can also be passed to `nanorc` at startup.
 - If one uses `nano04rc`, the thread pinning is enabled by default, and the arguments are the same before, but it won't `conf` if there isn't any `--thread-pinning-file` or `--disable-thread-pinning`.

I hope that's the behaviour we want?

File: src/nanorc/core.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 import copy as cp
 import json
 import os
+import copy as cp
```
Comment (alessandrothea) [inline src/nanorc/core.py:5]: Needed or leftover?

File: src/nanorc/cli.py [hunk 2]
```diff
@@ -19,6 +19,7 @@
 from . import __version__
 
 from anytree.resolver import Resolver
+from anytree import PreOrderIter
```
Comment (alessandrothea) [inline src/nanorc/cli.py]: needed?

File: src/nanorc/cli.py [hunk 3]
```diff
@@ -31,6 +32,7 @@
 from nanorc.core import NanoRC
 from nanorc.logbook import FileLogbook
 from nanorc.credmgr import credentials
+from nanorc.node import ApplicationNode
```
Comment (alessandrothea) [inline src/nanorc/cli.py]: needed?

File: setup.py [hunk 4]
```diff
@@ -12,6 +12,7 @@
         "graphviz",
         "anytree",
         "transitions",
+        "psutil",
```
Comment (alessandrothea) [inline setup.py]: What is this used for, eventually?

Comment (plasorak) [reply setup.py]: Well, this is really needed by `readout-affinity.py`, so it needs to be in the pyvenv. So it's not needed explicitly by nanorc, but we should make sure it's here.

File: src/nanorc/core.py (hunk 1 above)
Comment (plasorak) [reply src/nanorc/core.py:5]: Most of these import are useless indeed.

File: setup.py (hunk 4 above)
Comment (plasorak) [reply setup.py]: Actually thinking about this, this line should really go in readoutlibs

**Pull request merged**